### PR TITLE
auth: MoreCallCredentials.from(Credentials)

### PR DESCRIPTION
**Pull request by zhangkun83**
It converts Google Auth Library `Credentials` to `CallCredentials` and supersedes `ClientAuthInterceptor`, which is now deprecated.

### CODE REVIEW

File: auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java [hunk 1]
```diff
@@ -57,7 +57,10 @@
  *
  * <p>Uses the new and simplified Google auth library:
  * https://github.com/google/google-auth-library-java
+ *
+ * @deprecated use {@link GoogleAuthLibraryCallCredentials} instead.
  */
+@Deprecated
```
Comment (ejona86) [inline auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java]: Should we replace the code in this class with `next.newCall(method, callOptions.withCredentials(googleCredentials))`?

File: auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+/**
+ * Wraps {@link Credentials} as a {@link CallCredentials}.
+ */
+@ExperimentalApi("https//github.com/grpc/grpc-java/issues/1914")
+public final class GoogleAuthLibraryCallCredentials implements CallCredentials {
```
Comment (ejona86) [inline auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java]: This is quite verbose, but I don't really disagree with it. Due to dependencies, any other library we support will probably be in a different artifact. :-/

**Comment by ejona86** [comment]
@zhangkun83 LGTM, although we may want to discuss some the name GoogleAuthLibraryCallCredentials.

File: auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java (hunk 2 above)
Comment (ejona86) [inline auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java]: In offline discussion, we're thinking of going with `MoreCallCredentials.from(com.google.auth.Credential)`.

Comment (ejona86) [inline auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java]: So apparently C# is in a similar situation to us. I'm fine with copying:
https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Auth/GoogleGrpcCredentials.cs (http://www.grpc.io/grpc/csharp/html/N_Grpc_Auth.htm)

**Comment by ejona86** [comment]
@zhangkun83 LGTM

File: auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java (hunk 1 above)
Comment (zhangkun83) [inline auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java]: Done.

**Pull request merged**